collectors/AlertCollector.py

```diff
@@ -50,32 +55,26 @@ def collect(self):
             logger.warning(f'No alerts in the response for {self.name}. API code: {api_responding}')
             return
 
-        alert_labels = self.generate_alert_label_values(alerts)
-
-        for resource in alerts:
-            resource_id = resource.get('resourceId')
-            label_names = alert_metric._labelnames
+        for alert in alerts:
+            resource_id = alert.get('resourceId')
             labels = self.get_labels(resource_id, project_ids)
             if not labels:
                 continue
-            labels.extend([resource['alertDefinitionName'],
-                           resource['alertLevel'],
-                           resource['status'],
-                           resource["alertImpact"]])
-            for label_name, label_value in alert_labels.items():
-                alert_metric._labelnames += (label_name,)
-                labels.append(label_value)
-            alert_metric.add_metric(labels=labels, value=1)
-            alert_metric._labelnames = label_names
+            labels.extend([alert['alertDefinitionName'],
+                           alert['alertLevel'],
+                           alert['status'],
+                           alert["alertImpact"]])
+            alert_labels = self.generate_alert_label_values(alert)
+            alert_metric.add_metric(labels=labels, value=alert_labels)
```

Comment:
Solved quite simply.

Reply:
Just found it a minute ago. Totally makes sense then. Good catch!